src/spawn.coffee

```diff
@@ -30,12 +30,17 @@ procNotFoundError = (error, cmd) ->
   error.message = "Unable to find #{cmd}"
   error
 
-interpolatePort = (port) ->
-  (arg) ->
-    arg.replace '%port%', port
+processPorts = {}
+interpolatePorts = (args, processName, port) ->
+  processPorts["#{processName}.port"] = port
```

Comment:
This is approaching the point where it becomes an adhoc template language and stuff like this might be worth it:

```
stuff =
  port: port
  "#{processName}.port": otherPort

arg.replace /%([^%]+)%/, (m, key) ->
  if stuff[key] then stuff[key]
  else throw new Error "Unknown placeholder %#{key}%"
```

Otherwise we get into silent typos etc.